core/App/ehsm_provider.cpp

```diff
@@ -938,7 +940,26 @@ ehsm_status_t GenerateQuote(ehsm_data_t *quote)
     return EH_OK;
 }
 
+/*
+ *  @brief get mrenclave and mrsigner from quote
+ *  @param quote_data quote data
+ *  @param mr_signer_str the mrsigner
+ *  @param mr_enclave_str the mrenclave
+ */
+void get_mrenclave_mrsigner_from_quote(const uint8_t* quote_data, 
+                                       string &mr_signer_str, 
+                                       string &mr_enclave_str)
+{
+    sgx_quote3_t *p_sgx_quote = (sgx_quote3_t *)quote_data;
+    for(int i = 0; i < SGX_HASH_SIZE; i++ ) {
+        mr_enclave_str = mr_enclave_str.append(to_string(p_sgx_quote->report_body.mr_enclave.m[i]));
+        mr_signer_str = mr_signer_str.append(to_string(p_sgx_quote->report_body.mr_signer.m[i]));
+    }
```

Comment:
Agree. do not define as string.  
955-956 can use memcpy_s, e.g:
memcpy_s(&mr_enclave, sizeof(mr_enclave), p_sgx_quote->report_body.mr_enclave.m, sizeof(p_sgx_quote->report_body.mr_enclave.m))

Reply:
copy understood.